static/src/javascripts/projects/common/utils/robust.js

```diff
@@ -0,0 +1,35 @@
+/*jshint -W024 */
+define([
+    'raven',
+    'Promise'
+], function (
+    raven,
+    Promise
+) {
+
+    /*
+    * This purposefully does not return the Promise.
+    * It is designed to swallow and report on exceptions at a high level.
+    * If you need to act on the return value you are looking for a different abstraction
+    * */
+    function Robust(name, block, reporter) {
+
+        if (!reporter) {
+            reporter = raven.captureException;
+        }
+
+        new Promise(function (resolve, reject) {
+            try {
+                block();
+                resolve();
+            } catch (e) {
+                reject(e);
+            }
+        }).catch(function (e) {
+            reporter(e, { tags: { module: name } });
+        });
```

Comment:
what's the reason behind using a promise?

if `block` calls some async code, that wont be caught by the try/catch